app/models/nic/base.rb

```diff
@@ -223,6 +220,36 @@ def mac_uniqueness
       interface_attribute_uniqueness(:mac, Nic::Base.physical.is_managed)
     end
 
+    def update_domain_counters_on_create
+      return unless primary? && domain_id.present?
+      domain.increment!('total_hosts')
+    end
+
+    def update_domain_counters_on_update
+      primary_changed = previous_changes.include? 'primary'
+      domain_changed  = previous_changes.include? 'domain_id'
+      if primary_changed
+        if primary? #changed from non-primary to primary
+          domain.increment!('total_hosts') if domain_id.present?
+        else #changed from primary to non-primary
+          if domain_changed
+            Domain.find(previous_changes['domain_id'][0]).decrement!('total_hosts') if previous_changes['domain_id'][0].present?
```

Comment:
Does this need to be `Domain.unscoped.find`, to bypass taxonomies in the default scope?